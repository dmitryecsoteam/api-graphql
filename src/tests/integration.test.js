const graphqlServer = require('../server/server');
const mongo = require('../server/mongo');

const url = `http://localhost:${process.env.API_PORT}/${process.env.API_CONTEXT_ROOT}`

beforeAll(async () => {
    await graphqlServer.init();
    await mongo.connect();
});

test('should respond to originStartsWith query with capital letter', async () => {
    const injectOptions = {
        method: 'POST',
        url,
        headers: {
            'Content-Type': 'application/json'
        },
        payload: JSON.stringify({ query: '{originStartsWith(name: "T") { _id name_en name iata country_en } }' })
    };

    const response = await graphqlServer.server.inject(injectOptions);
    
    expect(response.result).toMatchSnapshot();
});

test('should respond to originStartsWith query with lowercase letter', async () => {
    const injectOptions = {
        method: 'POST',
        url,
        headers: {
            'Content-Type': 'application/json'
        },
        payload: JSON.stringify({ query: '{originStartsWith(name: "t") { _id name_en name iata country_en } }' })
    };

    const response = await graphqlServer.server.inject(injectOptions);
    
    expect(response.result).toMatchSnapshot();
});

test('should respond to destinationStartsWith query with capital letter', async () => {
    const injectOptions = {
        method: 'POST',
        url,
        headers: {
            'Content-Type': 'application/json'
        },
        payload: JSON.stringify({ query: '{destinationStartsWith(name: "O") { _id name_en name iata country_en } }' })
    };

    const response = await graphqlServer.server.inject(injectOptions);
    
    expect(response.result).toMatchSnapshot();
});

test('should respond to destinationStartsWith query with lowercase letter', async () => {
    const injectOptions = {
        method: 'POST',
        url,
        headers: {
            'Content-Type': 'application/json'
        },
        payload: JSON.stringify({ query: '{destinationStartsWith(name: "o") { _id name_en name iata country_en } }' })
    };

    const response = await graphqlServer.server.inject(injectOptions);
    
    expect(response.result).toMatchSnapshot();
});

test('should respond to destinationRating with all parameters passed', async () => {
    const injectOptions = {
        method: 'POST',
        url,
        headers: {
            'Content-Type': 'application/json'
        },
        payload: JSON.stringify({ query: '{destinationRating(museumRating: 3 zooAquaRating: 3 wellnessSpaRating: 3 mountainsRating: 3 beachRating: 3 foodRating: 3 shoppingRating: 3 historicalRating: 3 natureRating: 3) { _id name iata name_en country_en museumRating museumDescription zooAquaRating zooAquaDescription wellnessSpaRating wellnessSpaDescription mountainsRating mountainsDescription beachRating beachDescription foodRating foodDescription shoppingRating shoppingDescription historicalRating historicalDescription natureRating natureDescription } }' })
    };

    const response = await graphqlServer.server.inject(injectOptions);
    
    expect(response.result).toMatchSnapshot();
});

test('should respond to destination query', async () => {
    const injectOptions = {
        method: 'POST',
        url,
        headers: {
            'Content-Type': 'application/json'
        },
        payload: JSON.stringify({ query: '{destination(_id: 3) { _id name iata name_en country_en museumRating museumDescription zooAquaRating zooAquaDescription wellnessSpaRating wellnessSpaDescription mountainsRating mountainsDescription beachRating beachDescription foodRating foodDescription shoppingRating shoppingDescription historicalRating historicalDescription natureRating natureDescription } }' })
    };

    const response = await graphqlServer.server.inject(injectOptions);
    
    expect(response.result).toMatchSnapshot();
});

test('should respond to travel query', async () => {
    const injectOptions = {
        method: 'POST',
        url,
        headers: {
            'Content-Type': 'application/json'
        },
        payload: JSON.stringify({ query: '{travel(origin: 1, destination: 1, date: "2018-10-24") { _id destination { _id name_en country_en } date priceAirplane } }' })
    };

    const response = await graphqlServer.server.inject(injectOptions);
    
    expect(response.result).toMatchSnapshot();
});

test('should respond to travelFull query', async () => {
    const injectOptions = {
        method: 'POST',
        url,
        headers: {
            'Content-Type': 'application/json'
        },
        payload: JSON.stringify({ query: '{travelFull(_id: "5bbce95f81e97acfcd8e0a76") { _id destination { _id name_en country_en } date priceAirplane } }' })
    };

    const response = await graphqlServer.server.inject(injectOptions);

    expect(response.result).toMatchSnapshot();
});


afterAll(async () => {
    await mongo.close();
    await graphqlServer.server.stop();
    console.log("Server stopped");
});
