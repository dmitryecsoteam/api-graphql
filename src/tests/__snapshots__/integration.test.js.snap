// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should respond to destination query 1`] = `"{\\"data\\":{\\"destination\\":{\\"_id\\":3,\\"name\\":[\\"Nagoya\\",\\"Нагоя\\",\\"名古屋市\\",\\"NGO\\",\\"NKM\\"],\\"iata\\":\\"NGO\\",\\"name_en\\":\\"Nagoya\\",\\"country_en\\":\\"Japan\\",\\"museumRating\\":1,\\"museumDescription\\":\\"Ut tempor ex vel eleifend lobortis. Nunc vel tortor odio\\",\\"zooAquaRating\\":3,\\"zooAquaDescription\\":\\"Lorem ipsum dolor sit amet, consectetur adipiscing elit\\",\\"wellnessSpaRating\\":3,\\"wellnessSpaDescription\\":\\"Quisque scelerisque congue magna a vestibulum. Donec sed tincidunt purus\\",\\"mountainsRating\\":4,\\"mountainsDescription\\":\\"Pellentesque eu leo sed urna mattis suscipit eu eu tortor. Praesent auctor quam eu ex tristique pretium\\",\\"beachRating\\":0,\\"beachDescription\\":\\"Lorem ipsum dolor sit amet, consectetur adipiscing elit\\",\\"foodRating\\":2,\\"foodDescription\\":\\"Quisque scelerisque congue magna a vestibulum. Donec sed tincidunt purus\\",\\"shoppingRating\\":4,\\"shoppingDescription\\":\\"Lorem ipsum dolor sit amet, consectetur adipiscing elit\\",\\"historicalRating\\":0,\\"historicalDescription\\":\\"Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Aliquam tortor elit, vehicula ut enim pharetra, sagittis fermentum nulla\\",\\"natureRating\\":1,\\"natureDescription\\":\\"Ut tempor ex vel eleifend lobortis. Nunc vel tortor odio\\"}}}"`;

exports[`should respond to destinationRating with all parameters passed 1`] = `"{\\"data\\":{\\"destinationRating\\":[{\\"_id\\":5,\\"name\\":[\\"Nagasaki\\",\\"Нагасаки\\",\\"長崎市\\",\\"NGS\\"],\\"iata\\":\\"NGS\\",\\"name_en\\":\\"Nagasaki\\",\\"country_en\\":\\"Japan\\",\\"museumRating\\":4,\\"museumDescription\\":\\"Pellentesque eu leo sed urna mattis suscipit eu eu tortor. Praesent auctor quam eu ex tristique pretium\\",\\"zooAquaRating\\":5,\\"zooAquaDescription\\":\\"Pellentesque eu leo sed urna mattis suscipit eu eu tortor. Praesent auctor quam eu ex tristique pretium\\",\\"wellnessSpaRating\\":3,\\"wellnessSpaDescription\\":\\"Proin pharetra ante est, eget convallis magna tempor id. Etiam vitae aliquet massa\\",\\"mountainsRating\\":4,\\"mountainsDescription\\":\\"Quisque scelerisque congue magna a vestibulum. Donec sed tincidunt purus\\",\\"beachRating\\":3,\\"beachDescription\\":\\"Ut tempor ex vel eleifend lobortis. Nunc vel tortor odio\\",\\"foodRating\\":4,\\"foodDescription\\":\\"Proin pharetra ante est, eget convallis magna tempor id. Etiam vitae aliquet massa\\",\\"shoppingRating\\":3,\\"shoppingDescription\\":\\"Ut tempor ex vel eleifend lobortis. Nunc vel tortor odio\\",\\"historicalRating\\":5,\\"historicalDescription\\":\\"Quisque scelerisque congue magna a vestibulum. Donec sed tincidunt purus\\",\\"natureRating\\":5,\\"natureDescription\\":\\"Pellentesque eu leo sed urna mattis suscipit eu eu tortor. Praesent auctor quam eu ex tristique pretium\\"}]}}"`;

exports[`should respond to destinationStartsWith query with capital letter 1`] = `"{\\"data\\":{\\"destinationStartsWith\\":[{\\"_id\\":1,\\"name_en\\":\\"Osaka\\",\\"name\\":[\\"Osaka\\",\\"Осака\\",\\"大阪市\\",\\"KIX\\",\\"OSA\\",\\"ITM\\"],\\"iata\\":\\"OSA\\",\\"country_en\\":\\"Japan\\"},{\\"_id\\":4,\\"name_en\\":\\"Sapporo\\",\\"name\\":[\\"Sapporo\\",\\"Саппоро\\",\\"札幌市\\",\\"SPK\\",\\"CTS\\",\\"OKD\\"],\\"iata\\":\\"SPK\\",\\"country_en\\":\\"Japan\\"}]}}"`;

exports[`should respond to destinationStartsWith query with lowercase letter 1`] = `"{\\"data\\":{\\"destinationStartsWith\\":[{\\"_id\\":1,\\"name_en\\":\\"Osaka\\",\\"name\\":[\\"Osaka\\",\\"Осака\\",\\"大阪市\\",\\"KIX\\",\\"OSA\\",\\"ITM\\"],\\"iata\\":\\"OSA\\",\\"country_en\\":\\"Japan\\"},{\\"_id\\":4,\\"name_en\\":\\"Sapporo\\",\\"name\\":[\\"Sapporo\\",\\"Саппоро\\",\\"札幌市\\",\\"SPK\\",\\"CTS\\",\\"OKD\\"],\\"iata\\":\\"SPK\\",\\"country_en\\":\\"Japan\\"}]}}"`;

exports[`should respond to originStartsWith query with capital letter 1`] = `"{\\"data\\":{\\"originStartsWith\\":[{\\"_id\\":1,\\"name_en\\":\\"Tokyo\\",\\"name\\":[\\"Tokyo\\",\\"東京\\",\\"Токио\\",\\"TYO\\",\\"NRT\\",\\"HND\\"],\\"iata\\":\\"TYO\\",\\"country_en\\":\\"Japan\\"}]}}"`;

exports[`should respond to originStartsWith query with lowercase letter 1`] = `"{\\"data\\":{\\"originStartsWith\\":[{\\"_id\\":1,\\"name_en\\":\\"Tokyo\\",\\"name\\":[\\"Tokyo\\",\\"東京\\",\\"Токио\\",\\"TYO\\",\\"NRT\\",\\"HND\\"],\\"iata\\":\\"TYO\\",\\"country_en\\":\\"Japan\\"}]}}"`;

exports[`should respond to travel query 1`] = `"{\\"data\\":{\\"travel\\":{\\"_id\\":\\"5bbce95f81e97acfcd8e0a76\\",\\"destination\\":{\\"_id\\":1,\\"name_en\\":\\"Osaka\\",\\"country_en\\":\\"Japan\\"},\\"date\\":\\"2018-10-24\\",\\"priceAirplane\\":53}}}"`;
